sips/sip-022/sip-022-emergency-pox-fix.md

```diff
@@ -0,0 +1,284 @@
+`20e708e250bad3fb5d5ab84a70365c3c1cf0520c7a9f67cd5ff6b9fa94335ea5`.
+
+The immediate consequences for PoX of this bug being triggered are as follows:
+
+* The total STX locked to the PoX address owned by the caller of `stack-increase`
+  will be potentially higher than the amount of STX that the caller possesses.
+The caller will receive PoX payouts _as if_ they had locked that much STX.  So,
+the caller receives undue BTC.
+
+* The stacking threshold is raised to account for the PoX contract's reported
+  increase in STX locked.
+
+Similar consequences are expected when a stacker contributes to two different PoX addresses and 
+at least one PoX address would benefit from auto-unlocking. This behaviour has not been seen in the wild.
+
+
+Furthermore, if this bug is triggered enough times, the Stacks network will crash.  This is
+because of a runtime assertion in the PoX reward set calculation logic that
+verifies that the total locked STX does not exceed the total liquid STX.  This
+would no longer be true due to this bug.  The offending assertion is detailed
+below:
+
+```rust
+pub fn get_reward_threshold_and_participation(
+    pox_settings: &PoxConstants,
+    addresses: &[RawRewardSetEntry],
+    liquid_ustx: u128,
+) -> (u128, u128) {
+    let participation = addresses
+        .iter()
+        .fold(0, |agg, entry| agg + entry.amount_stacked);
+
+    assert!(
+        participation <= liquid_ustx,
+        "CORRUPTION: More stacking participation than liquid STX"
+    );
+```
+
+The `RawRewardSetEntry` data are pulled directly from the
+`reward-cycle-pox-address-list` data map, and the `.amount_stacked` field is
+equal to the `total-ustx` field that was erroneously set in `stack-increase`.
+
+Considering these consequences with respect to the [draft blockchain catastrophic
+failure and recovery guidelines](https://github.com/stacksgov/sips/pull/10),
+this bug would be categorized as requiring an _immediate_ hard fork to rectify.
+The network has not yet crashed, but it is in imminent danger of crashing and
+there is insufficient time to execute a Stacker-based SIP vote as has been customary for
+past hard forks.  This SIP instead proposes that this hard fork activate at the
+start of the next reward cycle (#58) at the time of this writing.
+
+There is less than one reward cycle remaining to fix this
+problem, and yet a Stacker vote would require at least one complete reward cycle
+to carry out vote (not accounting for any additional time required for
+sending out adequate public communications and tabulating the vote afterwards).
+A subsequent hard fork to re-enable PoX would have the effect of restoring
+Stacker voting.
+
+# Specification
+
+Given the lack of time to conduct a Stacker vote to activate this SIP, the
+proposed fix in this SIP is as parsimonious and discreet as possible.  It will
+execute as a set of **two hard forks**.
+
+The first hard fork, which activates at the start of reward cycle #58 will disable PoX.
```

Comment:
Should this say "before the start of reward cycle #58"?

Reply:
I think the wording is correct, since it will only activate when the cycle starts, but node operators will need to upgrade before the start of cycle 58

Reply:
In the activation section it says:

> The Bitcoin block activation height for the first hard fork will need to pass prior to the selection of the PoX anchor block for reward cycle #58 (Bitcoin block height 787851). This SIP proposes Bitcoin block height 787651, which is 200 Bitcoin blocks prior.

Reply:
I see your point, but i still feel the current wording is correct. 
the only alternative i can see working here is instead of 
>The first hard fork, which activates at the start of reward cycle #58 will disable PoX.
the bitcoin block where this fork is activated is mentioned

Reply:
How about:

> The first hard fork, which activates before the start of reward cycle #58, will disable PoX beginning with cycle #58.